### PR TITLE
Issue/84/follow redirects

### LINKED ISSUE #84: Fetching RSS at https://bialon.net/index.xml fails due to invalid xml.
Exception

    p.t.j.blog_posts.RssCheckingActor        : Exception for rss from Marcin Białoń 
        (http://bialon.net//index.xml): com.sun.syndication.io.ParsingFeedException: 
        Invalid XML: Error on line 6: The element type "hr" must be terminated 
        by the matching end-tag "</hr>".

Still RSS online validator shows that rss of this blog is valid. So we should find a work around for this issue.

Note: Same issue occurs for rss at http://luksza.org/feed/atom/

### PR DESCRIPTION
**Pull request by goostleek**
Fixes #84

### CODE REVIEW

**Comment by goostleek** [comment]
Ready for review.

File: src/test/groovy/pl/tomaszdziurko/jvm_bloggers/http/ProtocolSwitchingAwareConnectionRedirectHandlerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            TooManyRedirectsException e = thrown()
+    }
+
+    def "Should throw NPE on null connection parameter"() {
+        given:
+            @Subject tested = new ProtocolSwitchingAwareConnectionRedirectHandler(0);
+        when:
+            tested.handle(null, null)
+        then:
+            NullPointerException e = thrown()
```
Comment (tdziurko) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/http/ProtocolSwitchingAwareConnectionRedirectHandlerSpec.groovy]: Same situation here, 

    thrown(NullPointerException)

**Comment by tdziurko** [comment]
Looks great! There are small things to discuss, but overall everything is completely fine :)

**Comment by goostleek** [comment]
#TIL [#TIL](http://www.urbandictionary.com/define.php?term=TIL) :)

**Pull request merged**